RecipeMatch/Views/DraggableViewBackground.m

```diff
@@ -133,12 +146,38 @@ - (void)updateLikeCount:(DraggableView *)card{
 }
 
 // updates heart button for each card to show like status
-- (void)updateHeartBtn:(DraggableView *)nextCard{
+- (void)updateHeartBtn{
+    DraggableView *nextCard = [self->loadedCards objectAtIndex:0];
     [delegate checkLikeStatusFromDraggableViewBackground:nextCard withCompletion:^(BOOL succeeded, NSError * _Nullable error){
         if(succeeded){
-            [self->heartButton setImage:[UIImage imageNamed:@"heart-btn-filled"] forState:UIControlStateNormal];
+            [self->heartButton setImage:[UIImage imageNamed:HEART_FILL_IMG] forState:UIControlStateNormal];
         } else {
-            [self->heartButton setImage:[UIImage imageNamed:@"heart-btn"] forState:UIControlStateNormal];
+            [self->heartButton setImage:[UIImage imageNamed:HEART_IMG] forState:UIControlStateNormal];
+        }
+    }];
+}
+
+// updates save count for each card
+- (void)updateSaveCount{
+    DraggableView *card = [self->loadedCards objectAtIndex:0];
+    NSString *shortId = [(NSString *)card.recipeId substringFromIndex:51];
```

Comment:
also, reading the object id out of the URL, and assuming it will always be after 51 characters, isn't a good way - there are so many things that can go wrong.
you have the array of recipes at this point, and each recipe has it's id there... so should be pretty straight forward getting it from there. you just need a way to maintain the index of the current presented recipe, and you'll know how to find the current viewed recipe from the array.